tldw_Server_API/app/core/RAG/rag_service/rewrite_cache.py

```
@@ -20,6 +21,49 @@
 
 
 DEFAULT_PATH = Path("Databases/Rewrite_Cache/rewrite_cache.jsonl")
+_USER_DB_BASE = Path("Databases/user_databases")
+_ALLOWED_USER_ID_CHARS = set(string.ascii_letters + string.digits + "_-")
+
+
+def _is_relative_to(path: Path, base: Path) -> bool:
+    try:
+        path.relative_to(base)
+        return True
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)